src/plots/barPlot.ts

```diff
@@ -441,9 +441,20 @@ export module Plots {
             x += offset;
           }
 
+          var showLabel = true;
+          var labelPosition = {
+            x: x + w / 2 - measurement.width / 2,
+            y: positive ? y : y + h
+          };
+          if (labelPosition.x < 0 || labelPosition.x + measurement.width > +this._boundingBox.attr("width") ||
+              labelPosition.y < 0 || labelPosition.y + measurement.height > +this._boundingBox.attr("height")) {
```

Comment:
Are you sure this is correct in the case we have multiple datasets / stacked bars (not sure if those accept labels)